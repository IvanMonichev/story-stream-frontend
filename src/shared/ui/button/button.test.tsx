import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { describe, expect, it, vi } from 'vitest'
import Button from './button'

describe('Button component', () => {
  it('renders children correctly', () => {
    render(<Button>Click me</Button>)
    expect(screen.getByRole('button')).toHaveTextContent('Click me')
  })

  it('calls onClick handler when clicked', async () => {
    const handleClick = vi.fn()
    render(<Button onClick={handleClick}>Click</Button>)
    await userEvent.click(screen.getByRole('button'))
    expect(handleClick).toHaveBeenCalledTimes(1)
  })

  it('passes additional props to button element', () => {
    render(<Button type='submit'>Submit</Button>)
    expect(screen.getByRole('button')).toHaveAttribute('type', 'submit')
  })
  
  it('matches snapshot', () => {
    const { asFragment } = render(<Button>Snapshot</Button>)
    expect(asFragment()).toMatchSnapshot()
  })
})
