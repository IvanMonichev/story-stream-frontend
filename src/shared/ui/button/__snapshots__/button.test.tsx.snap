// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button component > matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="_button_b92fba"
  >
    Snapshot
  </button>
</DocumentFragment>
`;
